File: MSCL_Examples/Wireless/C++/MSCL_Examples_C++/NodePing/NodePing.cpp
//MSCL Example: NodePing
//	This example shows how to open a connection with a BaseStation,
//	ping a Node, and get the result and its information.
//
//Updated: 09/12/2014

#include <iostream>
using namespace std; 

#include "mscl/Communication/Connection.h"
#include "mscl/MicroStrain/Wireless/BaseStation.h"
#include "mscl/MicroStrain/Wireless/WirelessNode.h" 
#include "mscl/Exceptions.h"

int main(int argc, char **argv)
{
	//TODO: change these constants to match your setup
	const string COM_PORT = "COM3";
	const int NODE_ADDRESS = 31849;
	 
	try
	{
		//create a SerialConnection with the COM port
		mscl::Connection connection = mscl::Connection::Serial(COM_PORT);

		//create a BaseStation with the SerialConnection
		mscl::BaseStation baseStation(connection);

		//create a WirelessNode with the BaseStation
		mscl::WirelessNode node(NODE_ADDRESS, baseStation);

		cout << "Attempting to ping the Node..." << endl;

		//ping the Node
		mscl::PingResponse response = node.ping();

		//if the ping response was a success
		if(response.success()) 
		{
			//get some details from the response
			cout << "Successfully pinged Node " << NODE_ADDRESS << endl;
			cout << "Base Station RSSI: " << response.baseRssi() << endl;
			cout << "Node RSSI: " << response.nodeRssi() << endl;
		}
		else
		{
			cout << "Failed to ping Node " << NODE_ADDRESS << endl;
		}

	}
	catch(mscl::Error& e)
	{
		cout << "Error: " << e.what() << endl;
	}

	system("pause");
	return 0;
}

File: MSCL_Examples/Wireless/Python/NodePing.py
# MSCL Example: NodePing
#   This examples shows how to open a connection with a Base Station,
#   ping a Node, and get the result and its information
#
# Updated: 2/21/2024

import sys

# TODO: change this path to match your setup
# import the mscl library.
arch = sys.maxsize > 2**32 and "x64" or "x32"
sys.path.append("../../../Output/Python{sys.version_info.major}.{sys.version_info.minor}/{arch}/Release")

import mscl

# TODO: change these constants to match your setup
COM_PORT = "COM3"
NODE_ADDRESS = 65364

try:
    # create a Serial Connection with the specified COM Port, default baud rate of 921600
    connection = mscl.Connection.Serial(COM_PORT)

    # create a BaseStation with the connection
    baseStation = mscl.BaseStation(connection)

    # create a WirelessNode with the BaseStation we created
    node = mscl.WirelessNode(NODE_ADDRESS, baseStation)

    print("Attempting to ping the Node...")

    # ping the Node
    response = node.ping()

    # if the ping response was a success
    if response.success():
        # get some details from the response
        print("Successfully pinged Node", NODE_ADDRESS)
        print("Base Station RSSI:", response.baseRssi())
        print("Node RSSI:", response.nodeRssi())
    else:
        print("Failed to ping Node", NODE_ADDRESS)

except Exception as e:
    print("Error:", e    )
